Sources/SymbolKit/SymbolGraph/Symbol/FunctionSignature/FunctionParameter.swift

```diff
@@ -14,30 +14,66 @@ extension SymbolGraph.Symbol.FunctionSignature {
     /// An argument of a callable symbol.
     public struct FunctionParameter: Codable {
         enum CodingKeys: String, CodingKey {
-            case name
+            case externalName = "name"
+            case internalName
             case declarationFragments
             case children
         }
 
-        /// The name of the symbol, as referred to in user code (must match the name used in the documentation comment).
-        public var name: String // should we differentiate between internal and external names?
+        /// The name of the function parameter, as referred to in user code (must match the name used in the documentation comment).
+        public var name: String {
+            get { internalName }
+            set {
+                if _internalName == nil {
+                    externalName = newValue
+                } else {
+                    _internalName = newValue
+                }
+            }
+        }
+        
+        /// The external argument name of the function parameter.
+        public var externalName: String
+        
+        /// The internal parameter name of the function parameter.
```

Comment:
I would say that the "usual" name is the internal name. For example, this documentation comment 
```swift
/// Does something
/// - Parameters:
///   - first: Some number
///   - second: Some other number
```
could apply to either this Swift function
```swift
public func doSomething(with first: Int, and second: Int) {}
```
this C function
```c
void doSomething(int first, int second);
```
or this Objective-C method
```objc
- (void)doSomethingWith:(NSInteger)first and:(NSInteger)second;
```

C and Objective-C doesn't have separate external parameter names. In both those cases the `name` are simply "first", and "second".

I started looking into this because I was surprised that for that Swift declaration the function parameter JSON was
```json
{
  "name": "with",
  "internalName": "first",
  "declarationFragments": [
    ...
  ]
}
```
and
```json
{
  "name": "and",
  "internalName": "second",
  "declarationFragments": [
    ...
  ]
}
```

To me it would be more consistent with other languages if it was flipped so that the `"name"` was the internal name and the external name was encoded as "externalName". If you prefer I can make that change in the Swift symbol graph generator.

Reply:
I changed the API to `name` and `externalName` and updated both their documentation comments in https://github.com/apple/swift-docc-symbolkit/pull/64/commits/ead9b314b89b5edaad23235ea2a6c5ed628bc553.

How do you feel about that documentation?